app/controllers/application_controller.rb

```diff
@@ -20,6 +20,7 @@ def with_current_request
       if current.required_auth?
         raise Insights::API::Common::EntitlementError, "User not Entitled" unless check_entitled(current.entitlement)
 
+        Thread.current[:user] = UserContext.new(Insights::API::Common::Request.current!, params)
```

Comment:
Not needed for this PR, but any thoughts on a refactoring to move all the `Thread.current` logic into a User class.  Feels like something I've seen before.  😉

Reply:
@gmcculloug Thats where we started we had an cached instance variable in the pundit_user method in the controller, but that method cannot be accessed outside of the controller. The other challenge is reseting the user object at the end of the transaction. There is a User.with_user pattern which does something similar
https://github.com/ManageIQ/manageiq/blob/9d91ddfefcb3918192f3f5015d90c040a04e9a29/app/models/user.rb#L290
But we are already inside of a do block in the with_request
https://github.com/RedHatInsights/insights-api-common-rails/blob/0f6dd3f9c69693fd7289a4774096cc0a6e026f56/lib/insights/api/common/request.rb#L46
Not sure how 2 yield calls will work one inside the other.